google/cloud/internal/async_polling_loop.cc

```diff
@@ -56,9 +57,13 @@ class AsyncPollingLoopImpl
   }
 
   void DoCancel() {
-    if (op_.name().empty()) return;
     google::longrunning::CancelOperationRequest request;
-    request.set_name(op_.name());
+    {
+      std::unique_lock<std::mutex> lk(mu_);
+      canceled_ = true;
```

Comment:
I think we should set this just on `OnCancel()`: the cancel request might fail after all.

Reply:
Removed, because it looks like it was insufficient for the purpose, but I still think something should be done here ... otherwise, a cancel before `OnStart()` is triggered, and `op_name_` filled in, will be ignored.